src/metabase/api/pulse.clj

```diff
@@ -48,13 +48,14 @@
 
 (api/defendpoint POST "/"
   "Create a new `Pulse`."
-  [:as {{:keys [name cards channels skip_if_empty collection_id collection_position]} :body}]
+  [:as {{:keys [name cards channels skip_if_empty collection_id collection_position dashboard_id]} :body}]
   {name                su/NonBlankString
    cards               (su/non-empty [pulse/CoercibleToCardRef])
    channels            (su/non-empty [su/Map])
    skip_if_empty       (s/maybe s/Bool)
    collection_id       (s/maybe su/IntGreaterThanZero)
-   collection_position (s/maybe su/IntGreaterThanZero)}
+   collection_position (s/maybe su/IntGreaterThanZero)
+   dashboard_id        (s/maybe su/IntGreaterThanZero)}
```

Comment:
I don't know enough about our permissions system to know if this is safe. Is the following threat vector legitimate?

1) Someone has access to Dashboard 1 but cannot read Dashboard 2 at all
2) They create a new dashboard subscription that emails them, and hand-craft the POST to have a `dashboard_id` of `2`
3) They now receive daily emails with the contents of Dashboard 2, which they shouldn't have access to

Reply:
you definitely **have** to check dashboard perms before creating a subscription for it. Not sure if you need read perms for a Dashboard to create a subscription for it, or readwrite perms? I'd assume the later but check to be sure. You can do something like

```clj
(when dashboard_id
  (api/write-check Dashboard dashboard_id))
```

to check the perms